sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/JsonNodeUtils.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.azure.core.util.serializer.JsonNode;
+import com.google.gson.JsonElement;
+
+/**
+ * Helper methods for converting between Azure Core and GSON types.
+ */
+final class JsonNodeUtils {
+
+    /**
+     * Converts an Azure Core {@link JsonNode} into a GSON {@link JsonElement}.
+     *
+     * @param jsonNode The Azure Core {@link JsonNode}.
+     * @return The corresponding GSON {@link JsonElement}.
+     * @throws IllegalArgumentException If the {@link JsonNode} cannot be converted to a {@link JsonElement}.
+     */
+    public static JsonElement toGsonElement(JsonNode jsonNode) {
```

Comment:
I wish there is a good place to have these convertor `util` methods so that it's more discoverable. Not sure, may be as static in `GsonJsonSerializer` type, just a thought.

Reply:
This makes sense to me, right now these are package private. This could also be added to the interface in a more generic way, such as taking `Object` as the value type.